docs/enhancements/soft-install-timeout.md

```diff
@@ -0,0 +1,176 @@
+| Wait for finalizing        | installation (cluster)| controller | Assisted controller | hard coded |                       10h | Don't perform post install + don't send complete installation |
+| Wait for cluster operators | installation (cluster)| controller | Assisted controller | hard coded |                       10h | Don't perform rest of controller operations | Only CVO and console |
+| Add router CA              | installation (cluster)| controller | Assisted controller | hard coded |                       70m | Don't perform rest of controller operations |
+| Wait for OLM operators     | installation (cluster)| controller | Assisted controller || calculated from operators | | Don't perform rest of controller operations |
+| Apply manifests            | installation (cluster)| controller | Assisted controller | hard coded |                       10m | Don't perform rest of controller operations |
+| Wait for OLM operators CSV | installation (cluster)| controller | Assisted controller | | calculated from operators | Don't perform rest of controller operations |
+| Send complete installation | installation (cluster)| controller | Assisted controller | hard coded | 30m |  Don't perform rest of controller operations |
+
+
+There are 2 flows for completing installation.  One is managed by controller, and one by assisted service.
+
+In the flow managed by the assisted service, the following steps must be completed:
+
+- kubeconfig must be uploaded
+- cluster operators are successful
+- Monitored operators are either successful or failed
+
+In the flow managed by the controller all steps must be completed (in the table above).  The last step is a 
+a notification to complete installation.
+
+### Suggested changes
+- Only assisted service should notify installation completion
+- All steps by controller (as specified in the above table) will be known as stages in the assisted service.  This will enable the service to manage 
+them in a similar way that host stages are managed.
+- Controller will not stop activity due to timeout.
+- Assisted service should be able to terminate the controller.
+- All existing timeouts except installation timeout and rebooting timeout should be treated as soft timeouts (i.e cluster or host events)
+- Since OLM operators are not mandatory for successful installation, in case of timeout on these operators, the installation will proceed and marked as successul.
+- The suggested functionality should be optional. For SaaS it will be enabled globally by default and will be disabled by default
+for ZTP.  In addition, to use this feature it has to be enabled at organizational level. 
+- The 24h global installation timeout will be kept.  It will be considered as hard timeout. 
+
+### Open Questions
+
+- Should we collect must gather logs when the installation timeout expires, even
+  if we don't mark it as failed?
```

Comment:
What was the resolution here?

Reply:
Actually it was not handled

Reply:
So do we collect the must-gather at the 24 hour time out or not at all?

Reply:
If 24h timeout expires - the installation is marked as failed.  So it is handled the same way as failed installations are handled today.
The question is if we should collect must-gather that soft timeout expired and the user cancelled the installation.